src/p4est_communication.c

```diff
@@ -942,29 +942,22 @@ p4est_comm_checksum (p4est_t * p4est, unsigned local_crc, size_t local_bytes)
 #ifdef P4EST_ENABLE_MPI
   int                 mpiret;
   int                 p;
-  uint64_t            send[2];
-  uint64_t           *gather;
-
-  send[0] = (uint64_t) local_crc;
-  send[1] = (uint64_t) local_bytes;
-  gather = NULL;
-  if (p4est->mpirank == 0) {
-    gather = P4EST_ALLOC (uint64_t, 2 * p4est->mpisize);
-  }
-  mpiret = sc_MPI_Gather (send, 2, sc_MPI_LONG_LONG_INT,
-                          gather, 2, sc_MPI_LONG_LONG_INT, 0, p4est->mpicomm);
+  long long           send[2];
+  long long          *gather;
+
+  send[0] = (long long) local_crc;
+  send[1] = (long long) local_bytes;
+  gather = P4EST_ALLOC (long long, 2 * p4est->mpisize);
+  mpiret = sc_MPI_Allgather (send, 2, sc_MPI_LONG_LONG_INT,
```

Comment:
Would MPI_UNSIGNED_LONG_LONG be available? It may be a better fit.

Reply:
It was introduced in the MPI 2.0 standard.

Reply:
Ah ok then we don't use it.

Reply:
We should probably document that currently any higher bits than 32 of the inputs to the function are ignored.

Reply:
We do use MPI_UNSIGNED_LONG_LONG is libsc. Would we want to remove it, or should we configure test whether it exists?

Reply:
While we're at it, ar there any other MPI 2  types in our sc_mpi code?

Reply:
Let's keep this MPI type for the function (possibly just with a doc update on too-large values). The question of removing MPI_UNSIGNED_LONG_LONG belongs into libsc.

Reply:
That sounds good, should I create an issue for this in libsc?
Regarding the documentation, according to the C99 standard long long is at least 64 bits long. We might also change the parameters of `comm_checksum` to long long, although this might be inconsistent with the remaining input and output parameters.

Reply:
Yes that would be nice, and checking which other types are > MPI 1.3.

Let us just document in the .h files of p4est that we use the adler32 checksum on the lower 32 bits of the input variables. By the way, does adler32 survive more than 32 bits total input length, iow, does it roll over cleanly once we exceed that range? But this may also be left for a different issue in p4est.

Reply:
I'm leaning towards merging now and opening a p4est issue on 32 bit checksum issues.